### PR TITLE
MMT-3492: Creating resuable table component

### PR DESCRIPTION
**Pull request by mandyparson**

### CODE REVIEW

File: static/src/js/utils/stringifyCircularJSON.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const stringifyCircularJSON = (obj) => {
```
Comment (dmistry1) [inline static/src/js/utils/stringifyCircularJSON.js:1]: Add jsdoc that will describe what this util is doing

File: static/src/js/utils/stringifyCircularJSON.js
Comment (dmistry1) [inline static/src/js/utils/stringifyCircularJSON.js:1]: Add a test file

File: static/src/js/components/DraftList/DraftList.jsx [hunk 2]
```diff
@@ -53,6 +53,38 @@ const DraftList = ({ draftType }) => {
...
   }
 
+  // Building a Table using Data in items
+  const renderRows = (items.map((item) => {
+    const { conceptId, revisionDate, previewMetadata: { name, longName } } = item
+    const draftLink = `/drafts/${`${paramDraftType}`}/${conceptId}`
+
+    return (
+      [
+        <Link key={`${conceptId}/col1`} to={draftLink}>
```
Comment (cgokey) [inline static/src/js/components/DraftList/DraftList.jsx:63]: I think probably the keys can be included in the Table.jsx as well, this way you are just passing it a 2D array of data.

File: static/src/js/components/Table/Table.jsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}) => {
+  const renderHeaders = headers.map((header) => (
+    <th key={header}>{header}</th>
+  ))
+
+  const content = []
+
+  if (loading) {
+    content.push(
+      <For key="for-each-loading-key" each={[...new Array(5)]}>
```
Comment (cgokey) [inline static/src/js/components/Table/Table.jsx:30]: the constant 5 seems probablematic, can't that be headers.length?

File: static/src/js/components/Table/Table.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      const rowKey = stringifyCircularJSON(row)
+
+      return (
+        <tr key={`${rowKey}`}>
+          {
+            row.map((cell, index) => {
+              const cellKey = stringifyCircularJSON(cell)
+
+              return (
+                <td key={cellKey} className={classNames[index] || 'col-auto'}>
```
Comment (cgokey) [inline static/src/js/components/Table/Table.jsx:70]: yeah you are already providing a key here, so I don't think the key is needed in the caller.

File: static/src/js/components/DraftList/DraftList.jsx [hunk 5]
```diff
@@ -53,6 +53,38 @@ const DraftList = ({ draftType }) => {
     })
   }
 
+  // Building a Table using Data in items
+  const renderRows = (items.map((item) => {
```
Comment (cgokey) [inline static/src/js/components/DraftList/DraftList.jsx:57]: Call this `data`, as what you are building is a 2D array, you'll need to move the key into the Table component, but that gets the logic out of the caller.

Comment (mandyparson) [reply static/src/js/components/DraftList/DraftList.jsx:57]: let's huddle on this when you have time. I'm not sure which key you are referring to and I get eslint errors when I try to move keys to table

File: static/src/js/components/DraftList/DraftList.jsx [hunk 6]
```diff
@@ -53,6 +53,58 @@ const DraftList = ({ draftType }) => {
...
   }
 
+  // Building a Table using Data in items
+  const data = (items.map((item) => {
+    const { conceptId, revisionDate, previewMetadata: { name, longName } } = item
+    const draftLink = `/drafts/${`${paramDraftType}`}/${conceptId}`
+
+    return (
+      {
+        key: `${conceptId}`,
```
Comment (cgokey) [inline static/src/js/components/DraftList/DraftList.jsx:63]: you shouldn't need the backticks

**Review by cgokey — APPROVED** [review]

**Review by dmistry1 — APPROVED** [review]

**Pull request merged**